bin/get-video-s3-acls.py

```diff
@@ -0,0 +1,322 @@
+    try:
+        with open(cache_file, "r") as f_obj:
+            for line in f_obj.readlines():
+                (
+                    video_key,
+                    is_present_str,
+                    db_id_str,
+                    gloss_id_str,
+                    is_public_str,
+                ) = line.strip().split(CSV_DELIMITER)
+
+                is_present = is_present_str.strip().lower() == "true"
+                if is_present:
+                    nkeys_present += 1
+                    db_id = int(db_id_str)
+                    # Some don't have gloss_id's
+                    try:
+                        gloss_id = int(gloss_id_str)
+                    except ValueError:
+                        gloss_id = None
+                    is_public = is_public_str.strip().lower() == "true"
+                else:
+                    nkeys_absent += 1
+                    db_id = None
+                    gloss_id = None
+                    is_public = None
+
+                this_all_keys_dict[video_key] = [is_present, db_id, gloss_id, is_public]
+
+        print(f"PRESENT: {nkeys_present} keys", file=sys.stderr)
+        print(f"ABSENT:  {nkeys_absent} keys", file=sys.stderr)
+
+        return this_all_keys_dict
+
+    except FileNotFoundError:
+        print(f"File not found: {cache_file}", file=sys.stderr)
+        exit()
+
+
+# Get all keys from AWS S3
+def get_s3_bucket_raw_keys_list(
+    s3_bucket=AWS_S3_BUCKET, keys_file=S3_BUCKET_RAW_KEYS_FILE
+):
+    print(f"Getting raw AWS S3 keys recursively ({s3_bucket}) ...", file=sys.stderr)
+    with open(keys_file, "w") as f_obj:
+        subprocess.run(
+            [AWSCLIENT, "s3", "ls", f"s3://{s3_bucket}", "--recursive"],
+            env=NEW_ENV,
+            shell=False,
+            check=True,
+            text=True,
+            stdout=f_obj,
+        )
+
+    # Separate out just the key (also strip newline) from date, time, size, key
+    # Put the keys in an in-memory list
+    with open(keys_file, "r") as f_obj:
+        this_s3_bucket_raw_keys_list = [line.split()[3] for line in f_obj]
+    print(
+        f"{len(this_s3_bucket_raw_keys_list)} rows retrieved: {keys_file}",
+        file=sys.stderr,
+    )
+
+    # Write the keys back to the file, for cleanliness
+    with open(keys_file, "w") as f_obj:
+        for line in this_s3_bucket_raw_keys_list:
+            f_obj.write(f"{line}\n")
+
+    return this_s3_bucket_raw_keys_list
+
+
+# Get the video files info from NZSL Signbank
+def get_nzsl_raw_keys_dict(keys_file=NZSL_POSTGRES_RAW_KEYS_FILE):
+    this_nzsl_raw_keys_dict = {}
+    print(
+        f"Getting raw list of video file info from NZSL Signbank ...",
+        file=sys.stderr,
+    )
+    with open(keys_file, "w") as f_obj:
+        subprocess.run(
+            [
+                PGCLIENT,
+                "-t",
+                "-c",
+                "select id as db_id, gloss_id, is_public, videofile from video_glossvideo",
```

Comment:
suggestion: I think we can reduce our parsing logic by having postgres output in a more machine-readable format

I suspect there's an even better way, but you should be able to use [`COPY`](https://www.postgresql.org/docs/current/sql-copy.html) to output as e.g. a CSV:

```
COPY (SELECT ... FROM video_glossvideo LIMIT 5) TO stdout WITH CSV;
```

that should make it a lot easier to parse

Reply:
Ah yes, the old way was using `pg:psql` which offers very little control.
Now I'm using the native client we should have all the goodies back.

Reply:
I mean you should have been able to do this with `pg:psql` too since what we're doing here is native Postgres, and `pg:psql` is just piping through to that 🤷

Reply:
Done.
Wanted it to write directly to the file but suprocess.run chucks a wobbly. Can debug that another day.
For now it outputs to stdout as before, but the stdout is pre-formatted by postgres.
And yes, it's now just clean CSV and easy to parse!